back/apps/core-fca/src/dto/identity-for-sp.dto.ts

```diff
@@ -1,91 +1,27 @@
-import { Expose } from 'class-transformer';
-import {
-  IsAscii,
-  IsBoolean,
-  IsEmail,
-  IsOptional,
-  IsString,
-  MaxLength,
-  MinLength,
-} from 'class-validator';
+import { IsObject, IsString } from 'class-validator';
 
-import { IsPhoneNumberFca } from '../validators/is-phone-number-fca.validator';
+import { IsPhoneNumberSimpleValidator } from '../validators/is-phone-number-simple-validator.validator';
 import { IsSiret } from '../validators/is-siret-validator';
+import { IdentityFromIdpDto } from './identity-from-idp.dto';
 
-export class IdentityForSpDto {
-  @MinLength(1)
-  @IsAscii()
-  @Expose()
-  readonly sub!: string;
+export class IdentityForSpDto extends IdentityFromIdpDto {
+  @IsSiret({ groups: ['siret'] })
+  // By inheritance from IdentityFromIdpDto, siret is optional.
+  // We cannot override this optional behavior through class-transformer.
+  // Instead, siret is enforced as required via the sanitizer.
+  siret: string;
 
-  @IsString()
-  @MinLength(1)
-  @MaxLength(256)
-  @Expose()
-  readonly given_name: string;
-
-  @IsString()
-  @MinLength(1)
-  @MaxLength(256)
-  @Expose()
-  readonly usual_name: string;
-
-  @IsEmail()
-  @Expose()
-  readonly email: string;
-
-  @MinLength(1)
-  @IsAscii()
-  @Expose()
-  readonly uid: string;
-
-  /**
-   * @todo #484 Faire un validator pour siren
-   * @see https://gitlab.dev-franceconnect.fr/france-connect/fc/-/issues/484
-   */
-  @IsString()
-  @MinLength(1)
-  @MaxLength(256)
-  @IsOptional()
-  @Expose()
-  readonly siren?: string;
-
-  @IsSiret()
-  @Expose()
-  readonly siret: string;
+  @IsPhoneNumberSimpleValidator({ groups: ['phone_number'] })
+  phone_number?: string;
```

Comment:
For the phone number transformation, maybe we could do all operations inside the dto?

```
  @Transform(({ value }) => {
    // Use your custom phone number validation logic here
    const isValidPhoneNumber = IsPhoneNumberFca(value);
    return isValidPhoneNumber ? value : undefined;
  })
  @IsPhoneNumberFca()
  @IsOptional()
  @Expose()
  readonly phone_number?: string;

```